### PR TITLE
Adding Idempotent feature

### PR DESCRIPTION
**Pull request by ayush03agarwal**

### CODE REVIEW

**Comment by amarjeetsingh12** [comment]
LGTM

File: commons/src/main/scala/com/flipkart/connekt/commons/cache/DistributedCacheManager.scala [hunk 1]
```diff
@@ -27,6 +28,7 @@ object DistributedCacheManager extends CacheManager {
 
   private var cacheTTLMap: Map[DistributedCacheType.Value, CacheProperty] = Map[DistributedCacheType.Value, CacheProperty]()
   cacheTTLMap += DistributedCacheType.TransientUsers -> CacheProperty(0, 6.hours)
+  cacheTTLMap += DistributedCacheType.IdempotentCheck -> CacheProperty(0, 0.seconds)
```
Comment (kingster) [inline commons/src/main/scala/com/flipkart/connekt/commons/cache/DistributedCacheManager.scala]: not infinite cache, add some ttl to the entires. Maybe 24hrs

File: receptors/src/main/scala/com/flipkart/connekt/receptors/directives/IdempotentDirectives.scala [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import com.flipkart.metrics.Timed
+
+import scala.util.{Success, Try}
+
+trait IdempotentDirectives extends HeaderDirectives with Instrumented {
+
+  case class IdempotentRequestFailedRejection(transactionId: String) extends Rejection
+  type IsIdempotentReq = Boolean
+
+  val X_TRANSACTION_ID = "x-transaction-id"
```
Comment (kingster) [inline receptors/src/main/scala/com/flipkart/connekt/receptors/directives/IdempotentDirectives.scala]: This is not a transaction id, more like request-id

File: receptors/src/main/scala/com/flipkart/connekt/receptors/directives/IdempotentDirectives.scala [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  def isIdempotentRequest(appName: String): Directive0 = {
```
Comment (kingster) [inline receptors/src/main/scala/com/flipkart/connekt/receptors/directives/IdempotentDirectives.scala]: this is not a function, directives are mostly named as nouns

File: receptors/src/main/scala/com/flipkart/connekt/receptors/directives/IdempotentDirectives.scala [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import com.flipkart.connekt.commons.core.Wrappers.Try_#
+import com.flipkart.connekt.commons.metrics.Instrumented
+import com.flipkart.metrics.Timed
+
+import scala.util.{Success, Try}
+
+trait IdempotentDirectives extends HeaderDirectives with Instrumented {
+
+  case class IdempotentRequestFailedRejection(transactionId: String) extends Rejection
+  type IsIdempotentReq = Boolean
```
Comment (kingster) [inline receptors/src/main/scala/com/flipkart/connekt/receptors/directives/IdempotentDirectives.scala]: Why type aliasing for boolean?

**Comment by ayush03agarwal** [comment]
Updated PR

**Pull request merged**